### DISCUSSION TITLE
Loading Plugin when a certain directory is opened in nvim (on startup)

### DISCUSSION THREAD

**Discussion by josephcrawfordSRH** [LazyVim/LazyVim#2600] [Q&A] [answered]
I am looking to execute `nvim .` in a specific directory and only have a plugin loaded at that point.  I am trying to load `obsidian.nvim` when my directory of `~/path/to/Second Brain/` is opened in nvim.

I have done some research and found that I should probably be using the `cond` configuration option.  I have the following but it does not appear to load the plugin.

```
 cond = vim.fn["getcwd"] == vim.fn.expand("~") .. "/Second Brain/",
```

Can anyone provide me with some insight as to what I may be doing wrong?

**Comment by dpetka2001** [comment]
Try to execute in the directory that you want `:lua print(vim.fn.getcwd())` and see what it returns. I believe the trailing slash is not needed.

**Reply by josephcrawfordSRH** [reply]
Perfect that helped me determine the / was not required, the directories match now and it still does not load :/

Log Statements
```
print("getcwd: " .. vim.fn.getcwd())
print("expanded: " .. vim.fn.expand("~") .. "/Second Brain")
print(vim.fn.getcwd() == vim.fn.expand("~") .. "/Second Brain")
```
Output
```
getcwd: /Users/josephcrawford/Second Brain
expanded: /Users/josephcrawford/Second Brain
true
```

but obsidian.nvim is not loaded when I do `<leader>l`

Maybe `cond` is not what I want.  I noticed that when i open nvim in that directory the plugin is not loaded, however when I open nvim in any other directory that does not match the plugin is `disabled`

**Reply by dpetka2001** [reply, marked as answer]
This is working for me
```lua
return {
  {
    "epwalsh/obsidian.nvim",
    dependencies = {
      "nvim-lua/plenary.nvim",
    },
    cond = vim.fn.getcwd() == vim.fn.expand("~/Documents/Obsidian Vault"),
    opts = {
      workspaces = {
        {
          name = "personal",
          path = "~/Documents/Obsidian Vault",
        },
      },
    },
  },
}
```
You see that the plugin is `disabled`, because that's what `cond` does. It means it won't be loaded or lazy-loaded. When it's not `disabled`, then you can load it or lazy-load it, however you see fit.

**Reply by josephcrawfordSRH** [reply]
Interesting, it seems to have been because I also have `lazy = true` along with my `cond`.  I assumed I needed both.  I removed the lazy entry in the config and all is working.

Thank you very much for the assitance.

**Reply by Mo0rBy** [reply]
> This is working for me
> 
> ```lua
> return {
>   {
>     "epwalsh/obsidian.nvim",
>     dependencies = {
>       "nvim-lua/plenary.nvim",
>     },
>     cond = vim.fn.getcwd() == vim.fn.expand("~/Documents/Obsidian Vault"),
>     opts = {
>       workspaces = {
>         {
>           name = "personal",
>           path = "~/Documents/Obsidian Vault",
>         },
>       },
>     },
>   },
> }
> ```
> 
> You see that the plugin is `disabled`, because that's what `cond` does. It means it won't be loaded or lazy-loaded. When it's not `disabled`, then you can load it or lazy-load it, however you see fit.

How would do the same for any children within a directory?
For example, any children of `~/Documents/**`? Is it as simple as using a regex like I just demonstrated?

**Reply by Mo0rBy** [reply]
I figured out how:
```
  cond = function()
    local cwd = vim.fn.getcwd()
    local target = "/Users/myuser/your/target/path"
    return vim.startswith(cwd, target)
  end,

```
